api/src/org/apache/cloudstack/api/command/user/vm/DeployVMCmd.java

```diff
@@ -443,6 +449,37 @@ public String getKeyboard() {
         return dhcpOptionsMap;
     }
 
+    public Map<Long, DiskOffering> getDataDiskTemplateToDiskOfferingMap() {
+        if (diskOfferingId != null && dataDiskTemplateToDiskOfferingList != null) {
```

Comment:
if `dataDiskTemplateToDiskOfferingList == null or empty` and `diskOfferingId != null`?
We do not do anything in this case?

Reply:
In that cases, method returns an empty map

Reply:
Great. Then, what about changing things a little bit to reduce the number of lines inside an IF statement?

You could do `if(CollectionUtils.isEmpty(dataDiskTemplateToDiskOfferingList )){return new HashMap<Long, DiskOffering>()}`

Then, you would not need the `if` statement at line 457. This reduces one level of cyclomatic complexity in this method.

Reply:
@rafaelweingartner only the second part of the first if()s condition can be removed and the semantics woud slightly change as an exception instead of an empty map is really different.

Reply:
Actually, what I suggested is what is being done in the code now...
When I commented the code was different from its current state.